### ISSUE TITLE
Cannot build with @angular/cli@^1.6.1 because of webpack/jsdom bug

### ISSUE THREAD

**Issue by jskrzypek** [kangax/fabric.js#4585]
This still breaks with `@angular/cli@^1.6.1`, even with the #4576 fix. The issue is the `require('jsdom')` in the `HEADERS.js` file. Webpack will read this `require` and then fail because we don't get the `child_process` builtin in the build. We can't change the webpack config without using angular-cli's `eject` which would prevent us from using the `angular-cli` for our build process.

I think the best way to solve this would be to put [the jsdom part of `HEADERS.js`](https://github.com/kangax/fabric.js/blob/master/HEADER.js#L12-L25) in a separate file and build an additional `fabric.browser.js` and `fabric.browser.min.js` build that do not include this section.

Here's the webpack error. I'll make a PR in a couple hours.
```
ERROR in ./node_modules/jsdom/lib/jsdom/living/xmlhttprequest.js
Module not found: Error: Can't resolve 'child_process' in '/Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/lib/jsdom/living'
resolve 'child_process' in '/Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/lib/jsdom/living'
  Parsed request is a module
  using description file: /Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/package.json (relative path: ./lib/jsdom/living)
  after using description file: /Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/package.json (relative path: ./lib/jsdom/living)
    resolve as module
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/lib/jsdom/living/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/lib/jsdom/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/lib/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project/node_modules
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules/child_process)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process doesn't exist
        No description file found
        No description file found
        no extension
        .ts
        .js
        as directory
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project/node_modules
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules/child_process)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process doesn't exist
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./child_process)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/child_process doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/child_process.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/child_process.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/child_process doesn't exist
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./child_process)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/child_process doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/child_process.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/child_process.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/child_process doesn't exist
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/lib/jsdom/living/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/lib/jsdom/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/lib/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/jsdom/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/node_modules]
[/Users/jskrzypek/dev/node_modules]
[/Users/jskrzypek/node_modules]
[/Users/node_modules]
[/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process.ts]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process.js]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process.ts]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process.js]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/child_process]
[/Users/jskrzypek/dev/angular-fabric-project/child_process]
[/Users/jskrzypek/dev/angular-fabric-project/child_process.ts]
[/Users/jskrzypek/dev/angular-fabric-project/child_process.js]
[/Users/jskrzypek/dev/angular-fabric-project/child_process]
[/Users/jskrzypek/dev/angular-fabric-project/child_process]
[/Users/jskrzypek/dev/angular-fabric-project/child_process.ts]
[/Users/jskrzypek/dev/angular-fabric-project/child_process.js]
[/Users/jskrzypek/dev/angular-fabric-project/child_process]
 @ ./node_modules/jsdom/lib/jsdom/living/xmlhttprequest.js 4:22-46
 @ ./node_modules/jsdom/lib/jsdom/browser/Window.js
 @ ./node_modules/jsdom/lib/api.js
 @ ./node_modules/fabric/dist/fabric.js
 @ ./src/components/image-editor-dialog/base/canvas.ts
 @ ./src/components/image-editor-dialog/base/image-editor-dialog.component.ts
 @ ./src/components/image-editor-dialog/web/image-editor-dialog.component.ts
 @ ./src/components/web.ts
 @ ./src/common.web.module.ts
 @ ./src/common.web.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi ./src/main.ts
ERROR in ./node_modules/sntp/lib/index.js
Module not found: Error: Can't resolve 'dgram' in '/Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/lib'
resolve 'dgram' in '/Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/lib'
  Parsed request is a module
  using description file: /Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/package.json (relative path: ./lib)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: /Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/package.json (relative path: ./lib)
    resolve as module
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/lib/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project/node_modules
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules/dgram)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram doesn't exist
        No description file found
        Field 'browser' doesn't contain a valid alias configuration
        No description file found
        no extension
          Field 'browser' doesn't contain a valid alias configuration
        .ts
          Field 'browser' doesn't contain a valid alias configuration
        .js
          Field 'browser' doesn't contain a valid alias configuration
        as directory
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project/node_modules
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules/dgram)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram doesn't exist
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./dgram)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dgram doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dgram.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dgram.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/dgram doesn't exist
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./dgram)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dgram doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dgram.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dgram.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/dgram doesn't exist
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/lib/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/node_modules]
[/Users/jskrzypek/dev/node_modules]
[/Users/jskrzypek/node_modules]
[/Users/node_modules]
[/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram.ts]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram.js]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram.ts]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram.js]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dgram]
[/Users/jskrzypek/dev/angular-fabric-project/dgram]
[/Users/jskrzypek/dev/angular-fabric-project/dgram.ts]
[/Users/jskrzypek/dev/angular-fabric-project/dgram.js]
[/Users/jskrzypek/dev/angular-fabric-project/dgram]
[/Users/jskrzypek/dev/angular-fabric-project/dgram]
[/Users/jskrzypek/dev/angular-fabric-project/dgram.ts]
[/Users/jskrzypek/dev/angular-fabric-project/dgram.js]
[/Users/jskrzypek/dev/angular-fabric-project/dgram]
 @ ./node_modules/sntp/lib/index.js 5:14-30
 @ ./node_modules/hawk/lib/index.js
 @ ./node_modules/request/request.js
 @ ./node_modules/request/index.js
 @ ./node_modules/jsdom/lib/jsdom/browser/resource-loader.js
 @ ./node_modules/jsdom/lib/api.js
 @ ./node_modules/fabric/dist/fabric.js
 @ ./src/components/image-editor-dialog/base/canvas.ts
 @ ./src/components/image-editor-dialog/base/image-editor-dialog.component.ts
 @ ./src/components/image-editor-dialog/web/image-editor-dialog.component.ts
 @ ./src/components/web.ts
 @ ./src/common.web.module.ts
 @ ./src/common.web.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi ./src/main.ts
ERROR in ./node_modules/sntp/lib/index.js
Module not found: Error: Can't resolve 'dns' in '/Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/lib'
resolve 'dns' in '/Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/lib'
  Parsed request is a module
  using description file: /Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/package.json (relative path: ./lib)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: /Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/package.json (relative path: ./lib)
    resolve as module
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/lib/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/angular-fabric-project/node_modules/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/dev/node_modules doesn't exist or is not a directory
      /Users/jskrzypek/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project/node_modules
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules/dns)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dns doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dns.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dns.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dns doesn't exist
        No description file found
        Field 'browser' doesn't contain a valid alias configuration
        No description file found
        no extension
          Field 'browser' doesn't contain a valid alias configuration
        .ts
          Field 'browser' doesn't contain a valid alias configuration
        .js
          Field 'browser' doesn't contain a valid alias configuration
        as directory
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project/node_modules
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./node_modules/dns)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dns doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dns.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dns.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/node_modules/dns doesn't exist
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./dns)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dns doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dns.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dns.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/dns doesn't exist
      looking for modules in /Users/jskrzypek/dev/angular-fabric-project
        using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: .)
          using description file: /Users/jskrzypek/dev/angular-fabric-project/package.json (relative path: ./dns)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dns doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dns.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/jskrzypek/dev/angular-fabric-project/dns.js doesn't exist
            as directory
              /Users/jskrzypek/dev/angular-fabric-project/dns doesn't exist
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/lib/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/sntp/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/node_modules]
[/Users/jskrzypek/dev/node_modules]
[/Users/jskrzypek/node_modules]
[/Users/node_modules]
[/node_modules]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dns]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dns.ts]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dns.js]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dns]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dns]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dns.ts]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dns.js]
[/Users/jskrzypek/dev/angular-fabric-project/node_modules/dns]
[/Users/jskrzypek/dev/angular-fabric-project/dns]
[/Users/jskrzypek/dev/angular-fabric-project/dns.ts]
[/Users/jskrzypek/dev/angular-fabric-project/dns.js]
[/Users/jskrzypek/dev/angular-fabric-project/dns]
[/Users/jskrzypek/dev/angular-fabric-project/dns]
[/Users/jskrzypek/dev/angular-fabric-project/dns.ts]
[/Users/jskrzypek/dev/angular-fabric-project/dns.js]
[/Users/jskrzypek/dev/angular-fabric-project/dns]
 @ ./node_modules/sntp/lib/index.js 6:12-26
 @ ./node_modules/hawk/lib/index.js
 @ ./node_modules/request/request.js
 @ ./node_modules/request/index.js
 @ ./node_modules/jsdom/lib/jsdom/browser/resource-loader.js
 @ ./node_modules/jsdom/lib/api.js
 @ ./node_modules/fabric/dist/fabric.js
 @ ./src/components/image-editor-dialog/base/canvas.ts
 @ ./src/components/image-editor-dialog/base/image-editor-dialog.component.ts
 @ ./src/components/image-editor-dialog/web/image-editor-dialog.component.ts
 @ ./src/components/web.ts
 @ ./src/common.web.module.ts
 @ ./src/common.web.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi ./src/main.ts
```

**Comment by asturur** [comment]
Is unclear to me if the fix #4576 is uneffective or is uneffective in conjunction with angular tools.

**Comment by jskrzypek** [comment]
It's not effective in conjunction with angular tools, I haven't tested with plain webpack and babel, that's why I made another issue separate from #4516.

**Comment by jskrzypek** [comment]
After Looking at it, I think it would actually be simplest to just create a separate browser build by cutting out the node/jsdom part of `HEADERS.js` rather than starting to split it out into separate files, similar to what is done with the `noSVGExport` and `noSVGImport` comments.

**Comment by asturur** [comment]
I can wrap the code in a reg_ex comment and cut it out, but i do not want to output 2 separate builds now.
I believe this is one problem but the same problem can happen to many other packages. why the angular tools can't be improved to allow configuration?

**Comment by jskrzypek** [comment]
> why the angular tools can't be improved to allow configuration?

That's a looooong debate 😄

However, I realize now that the `browser` field spec seems to be something that was working before the @angular/cli@1.6.1 release and that something in that change broke their support for that feature of webpack. 

If it's ok with you, I'd like to leave this issue open a bit longer while I look into this. If it does turn out it's actually their bug (which now seems likely to me) then I'll close this issue later today and point to the discussion on their repo.

**Comment by asturur** [comment]
i m fine with leaving the issue open and discuss it of course.

**Comment by jskrzypek** [comment]
Ok I looked into this and @angular/cli's @ngtools/webpack provides a `paths-plugin` for webpack that uses typescript's path mapping resolution for using path aliases. In @angular/cli@1.6.1 they added a change that applied this plugin to `imports` from .js files in addition to .ts files. 

This caused our paths specification to suddenly be applied to our import of `fabric.js` as well as our own internal code. However typescript resolution does not respect the `browser` field in the `package.json` file, so suddenly all of fabric's internal requires were being pared and resolved by webpack, meaning that jsdom et al. get imported instead of ignored as per the normal webpack behavior. (This is a bit related to this issue: https://github.com/angular/angular-cli/issues/7540)

This turned out not to be a bug but rather a poorly documented interaction of features. When I changed the `paths` specification in our tsconfig file to be more selective and only apply to the import paths we actually need, the build with fabric worked.

Thank you for the quick response and sorry for any headaches I may have caused!
